### PR TITLE
Fix ListBox typeahead spacebar issue

### LINKED ISSUE #851: ListBox typeahead shouldn't select the focused item when pressing space
# 🐛 Bug Report 

Listbox supports typing to search for items in the list. A problem arises however when a user searches for an item with a space in its name (e.g. "Item 10") since hitting the space key will select the currently focused item. 

## 🤔 Expected Behavior

When the user is typing to search through items in the listbox, hitting space shouldn't select the focused item. Instead, typeahead should continue to work as expected and focus should be moved to the first item that matches what the user has typed.

## 😯 Current Behavior

As mentioned above, hitting space selects the currently focused item, which is a problem if the user attempts to type to search for a item with a space in its name.

If you build storybook locally, this story is a good place to reproduce: http://localhost:9003/?path=/story/listbox--listbox-w-many-sections-and-selection.

## 💁 Possible Solution

Possibly add a timeout that prevents space from selecting the focused item while the user types in the ListBox. Stopping propagating in useTypeSelect + changing it to be a capturing listener might be sufficient.

code for typeahead: https://github.com/adobe/react-spectrum/blob/main/packages/@react-aria/selection/src/useTypeSelect.ts

## 🌍 Your Environment

| Software         | Version(s) |
| ---------------- | ---------- |
| react-spectrum   |  3.1.0
| Browser          |    All
| Operating System |  All

## 🧢 Your Company/Team

RSP

### PR DESCRIPTION
**Pull request by solimant**
Closes #851

## ✅ Pull Request Checklist:

- [X] Included link to corresponding [React Spectrum GitHub Issue](https://github.com/adobe/react-spectrum/issues).
- [X] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [X] Filled out test instructions.
- [ ] Updated documentation (if it already exists for this component).
- [X] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/TR/wai-aria-practices-1.1/)

## 📝 Test Instructions:
1. Go to storybook
2. Go to **ListBox > ListBox w/ many sections and selection**
3. Make sure the listbox is in focus
4. Type in an item name including the space character; e.g. `Section 8`
5. If the item is highlighted without being selected, tests 1/2 pass ✅ 
6. Wait about a second or so (to clear the typeahead timeout), then press the *Spacebar* on your keyboard
7. If the item is selected, tests 2/2 pass ✅ 
8. Everything is fine, nothing is ruined
## 🧢 Your Project:
Adobe React-Spectrum
rsp

### CODE REVIEW

File: packages/@react-spectrum/listbox/test/ListBox.test.js [hunk 1]
```diff
@@ -489,6 +493,34 @@ describe('ListBox', function () {
       expect(document.activeElement).toBe(options[4]);
     });
 
+    it('supports the space character in a search', function () {
```
Comment (snowystinger) [inline packages/@react-spectrum/listbox/test/ListBox.test.js:496]: if you use jest.useFakeTimers, you can step through time to verify that you can use space bar to select the option after the typeahead timeout has completed, i won't require that you write that test, just an interesting thing you can do

File: packages/@react-aria/selection/src/useTypeSelect.ts [hunk 2]
```diff
@@ -88,6 +99,8 @@ function getStringForKey(key: string) {
   // See https://www.w3.org/TR/uievents-key/
   if (key.length === 1 || !/^[A-Z]/i.test(key)) {
     return key;
+  } else if (key === 'Spacebar') { // For IE 11
```
Comment (snowystinger) [inline packages/@react-aria/selection/src/useTypeSelect.ts:102]: we don't care about IE 11 anymore if you want to remove this, no sense in dead code :)

File: packages/@react-spectrum/listbox/test/ListBox.test.js (hunk 1 above)
Comment (solimant) [reply packages/@react-spectrum/listbox/test/ListBox.test.js:496]: Thanks, @snowystinger, good idea! I called it out in the test instructions, so I think I should include it as well as a test.

File: packages/@react-aria/selection/src/useTypeSelect.ts (hunk 2 above)
Comment (solimant) [reply packages/@react-aria/selection/src/useTypeSelect.ts:102]: bye_ie

File: packages/@react-spectrum/listbox/test/ListBox.test.js (hunk 1 above)
Comment (solimant) [reply packages/@react-spectrum/listbox/test/ListBox.test.js:496]: Done.

**Pull request merged**